src/ape/cli/choices.py

```diff
@@ -313,7 +313,16 @@ def convert(self, value: Any, param: Optional[Parameter], ctx: Optional[Context]
             # By-pass choice constraints when using adhoc network
             return value
 
-        return super().convert(value, param, ctx)
+        try:
+            return super().convert(value, param, ctx)
+        except BadParameter:
+            # Find out actual bad parts of the value to show better error.
+            # The following line should raise a nicer error.
+            networks.get_provider_from_choice(network_choice=value)
+
+            # If an error was not raised for some reason, raise the original
+            # BadParameter error.
+            raise
```

Comment:
I'm okay with this, but there's not a cleaner option?

Reply:
you don' like `click.exceptions.BadParameter`? This is what it was raising before.

Reply:
imo this is like the cleaniest way of handling this edge case that likely will never happen anyway, lol.

Reply:
I changed it to raise a new error without all the network options, but still keeping it BadParameter and also note this edge case i dont think is possible to occur

Reply:
for some reason github is being weird and not showing the commits